src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/InternationalPostalService.java

```diff
@@ -0,0 +1,216 @@
+        Address applicantAddress = getAddress(caseData.get("applicantAddress"));
+
+        List<String> errors = new ArrayList<>();
+        if (applicantOutsideUK.equals("Yes")
+            && applicantAddress != null
+            && ObjectUtils.isEmpty(applicantAddress.getCountry())) {
+            errors.add("If applicant resides outside of UK, Please provide the country of residence.");
+        }
+
+        String respondentOutsideUK = String.valueOf(caseData.get(RESPONDENT_RESIDE_OUTSIDE_UK));
+        Address respondentAddress = getAddress(caseData.get("respondentAddress"));
+
+        if (respondentOutsideUK.equals("Yes")
+            && respondentAddress != null
+            && ObjectUtils.isEmpty(respondentAddress.getCountry())) {
+            errors.add("If respondent resides outside of UK, Please provide the country of residence.");
+        }
+        return errors;
+    }
+
+    /**
+     * Determines whether the recipient of a letter resides outside of the UK based on the given case data and recipient.
+     *
+     * @param caseData The case data as a map of key-value pairs.
+     * @param recipient The recipient of the letter.
+     * @return True if the recipient resides outside of the UK, false otherwise.
+     */
+    public boolean isRecipientResideOutsideOfUK(Map<String, Object> caseData, String recipient) {
+        String addressee = ObjectUtils.isNotEmpty(recipient) ?  WordUtils.capitalizeFully(recipient) : recipient;
+        boolean isInternational;
+        switch (addressee) {
+            case APPLICANT -> isInternational = isApplicantResideOutsideOfUK(caseData);
+            case RESPONDENT -> isInternational = isRespondentResideOutsideOfUK(caseData);
+            case INTERVENER1 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener1Wrapper(caseData.get(INTERVENER_ONE));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER2 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener2Wrapper(caseData.get(INTERVENER_TWO));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER3 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener3Wrapper(caseData.get(INTERVENER_THREE));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER4 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener4Wrapper(caseData.get(INTERVENER_FOUR));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            default -> isInternational = false;
+        }
+        return isInternational;
+    }
+
+    public boolean isRecipientResideOutsideOfUK(FinremCaseData caseData, String recipient) {
+        String party = String.valueOf(recipient);
```

Comment:
Is this necessary?

Reply:
Yes, Just in case recipient is null. Mostly for test.

Reply:
If recipient is null then it makes more sense to return early. Otherwise your code does odd things like set party ="null" and then convert it to "Null" and then do a switch on it.

Reply:
public static String valueOf(Object obj) {
        return (obj == null) ? "null" : obj.toString();
    }
so it will go to default

Reply:
covered in test as well.